tiny-event-sourcing-lib/src/main/kotlin/ru/quipy/core/EventSourcingService.kt

```diff
@@ -184,11 +192,19 @@ class EventSourcingService<ID : Any, A : Aggregate, S : AggregateState<ID, A>>(
             throw e
         }
 
+        if (eventSourcingProperties.sagasEnabled) {
+            sagaContext.causationId = sagaContext.currentEventId
+            sagaContext.currentEventId = UUID.randomUUID()
```

Comment:
why random?

Reply:
We can try using event id
But it seems complicated